Dalamud/Interface/ImGuiNotification/Notification.cs

```diff
@@ -32,7 +32,33 @@ public sealed record Notification : INotification
     public INotificationIcon? Icon { get; set; }
 
     /// <inheritdoc/>
-    public ISharedImmediateTexture? IconTexture { get; set; }
+    public ISharedImmediateTexture? ImmediateIconTexture { get; set; }
+
+    /// <inheritdoc/>
+    public IDalamudTextureWrap? IconTexture
+    {
+        get => this.ImmediateIconTexture?.GetWrapOrDefault();
+        set => this.ImmediateIconTexture = value != null ? new ForwardingSharedImmediateTexture(value) : null;
+    }
+
+    /// <inheritdoc/>
+    public Task<IDalamudTextureWrap?>? IconTextureTask
+    {
+        get => Task.FromResult(this.ImmediateIconTexture?.GetWrapOrDefault());
+
+        set
+        {
+            if (value == null)
+            {
+                this.ImmediateIconTexture = null;
+            }
+            else
+            {
+                var dalamudTextureWrap = value.Result;
```

Comment:
Assigning failed tasks should not throw. IMO it's better to just leave previous properties/methods alone and add ISIT on its own, and only on drawing refer to ISIT at first and then check the rest to figure out the correct thing to draw.

Reply:
I'd basically need to throw out everything I've already done if that's the route you want to go, as it'd be faster for me to start on a new branch and implement a ImmediateIconTexture there rather than trying to put back in what I've already removed

Have added in a try/catch, not pretty but does aleviate that above concern

Reply:
Hopefully nobody refers to the state of the task after assigning it to the notification, too. Not sure why would one do that, so I'd guess it's be safe for this property to be "inconsistent".

Reply:
Only thing I could think of is creating a new Task.FromException<IDalamudTextureWrap?>(exception), storing it in a field and returning that when IconTextureTask.get is called

Then when a set resolves correctly, it sets the failed task field to null.

Not sure why would want to do it either, if I don't do something like above, it shouldn't cause a crash as they'll still have to null check, just a bit incosnsistent as you said

Reply:
Another possible way would be making ForwardingSIT store Task<IDTW>  instead?

Reply:
I've considered it however it'd require the GetWrapOrEmpty function to call `Service<DalamudAssetManager>.Get().Empty4X4` which adds a dependency on an internal service.

Now if we made it so you could only get a ForwardingSharedImmediateTexture's from the ITextureProvider and it injected the 4x4 into the ForwardingSharedImmediateTexture that'd be fine but it adds a hard dependency to an object that could concievably be constructed outside of Dalamud